SIPS/sip-30.md

```diff
@@ -0,0 +1,248 @@
+
+### Snap Manifest
+
+No changes are required to the snap manifest.
+
+### Client Implementation
+
+#### Entropy Sources
+
+If a snap requests a list of available entropy sources, and it has the permission to do so, the wallet MUST return a list of `EntropySource` objects.
+
+The client MUST have a primary entropy source, which is used when no source is specified. In the list of available entropy sources, the primary source MUST be marked as `primary: true`.
+
+#### Handling Entropy Requests
+
+If a snap requests entropy and includes the `source` parameter for an entropy source of type `mnemonic`, the wallet MUST return entropy corresponding to that source, if it exists.
+
+If the source does not exist, the wallet MUST respond with an error.
+
+If the request does not include the `source` parameter, the wallet MUST return entropy from the primary source.
+
+#### Creating Accounts
+
+A client  MAY invoke the `keyring.createAccount` method with an `entropySource` parameter in the `options` object.
+
+The `entropySource` parameter MUST be a string which uniquely identifies the entropy source to use. It is not guaranteed to be the same string visible to any other snap, but should always refer to the same source in the context of interactions between the snap and the client.
+
+### Snap Implementation
+
+If a snap is asked to create an account via `keyring.createAccount`, and the `entropySource` parameter is provided, and the snap requires entropy to create an account, the snap SHOULD request the entropy from the specified source.
+
+### New RPC Methods
+
+#### `snap_listEntropySources`
+
+The method returns an array of `EntropySource` objects, each representing an available entropy source. The Snap MAY choose to display this list to the user.
+
+```typescript
+const entropySources = await snap.request({
+  method: "snap_listEntropySources",
+});
+// [
+//   { name: "Phrase 1", id: "phrase-1" },
+//   { name: "Phrase 2", id: "phrase-2" },
+// ]
+```
+
+### Existing RPC Methods
+
+#### `snap_getEntropy`
+
+##### Parameters
+An object containing:
+
+- `version` - The number 1
+- `salt` (optional) - An arbitrary string to be used as a salt for the entropy. This can be used to generate different entropy for different purposes.
+- `source` (optional) - The ID of the entropy source to use. If not specified, the default entropy source will be used.
+
+#### Returns
+The entropy as a hexadecimal string.
+
+#### Example
+
+```typescript
+const entropy = await snap.request({
+  method: "snap_getEntropy",
+  params: {
+    version: 2,
+    salt: "my-salt",
+    source: "1234-5678-9012-3456-7890",
+  },
+});
+// '0x1234567890abcdef'
+```
+
+#### `snap_getBip32Entropy`
+
+##### Parameters
+
+- `path`: An array starting with `m` containing the BIP-32 derivation path of the key to retrieve.
+- `source`: The ID of the entropy source to use.
```

Comment:
Can we standardize the format for these? E.g. mark as optional etc as done for `snap_getEntropy`